PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java

```diff
@@ -0,0 +1,296 @@
+
+    private static final String GUID_REGEX = "^[{]?[0-9a-fA-F]{8}" + "-([0-9a-fA-F]{4}-)" + "{3}[0-9a-fA-F]{12}[}]?$";
+
+    protected ServicePrincipalLoginDialog(@Nullable Project project) {
+        super(project, true);
+        this.project = project;
+        init();
+        $$$setupUI$$$(); // tell IntelliJ to call createUIComponents() here.
+        this.setTitle("Sign in Service Principal");
+        super.setOKButtonText("Sign In");
+
+        pasteFromClipboard();
+        uiTextComponents2Json();
+
+        // initialize cert file select
+        FileChooserDescriptor pem = FileChooserDescriptorFactory.createSingleFileDescriptor("pem");
+        pem.withFileFilter(file -> StringUtils.equalsIgnoreCase(file.getExtension(), "pem"));
+        certFileTextField.addActionListener(new ComponentWithBrowseButton.BrowseFolderActionListener<>("Select Certificate File", null, certFileTextField, null,
+                pem, TextComponentAccessor.TEXT_FIELD_WHOLE_TEXT) {
+            @Nullable
+            protected VirtualFile getInitialFile() {
+                return LocalFileSystem.getInstance().findFileByPath(FileUtil.getDirectoryWithinUserHome("/").toString());
+            }
+        });
+
+        Stream.of(clientIdTextField, tenantIdTextField, keyPasswordField, certFileTextField.getTextField()).map(JTextComponent::getDocument)
+                .forEach(document -> document.addDocumentListener(new TextDocumentListenerAdapter() {
+                    @Override
+                    public void onDocumentChanged() {
+                        uiTextComponents2Json();
+                    }
+                }));
+
+        this.certificateRadioButton.addActionListener((e) -> uiTextComponents2Json());
+        this.passwordRadioButton.addActionListener((e) -> uiTextComponents2Json());
+
+        this.jsonDataEditor.addDocumentListener(new DocumentListener() {
+            @Override
+            public void documentChanged(@NotNull DocumentEvent event) {
+                if (jsonDataEditor.getText().equals(memo)) {
+                    return;
+                }
+                json2UIComponents(jsonDataEditor.getText());
+            }
+        });
+        certificateRadioButton.addActionListener(e -> syncComponentStatusWhenRadioButtonChanges());
+        passwordRadioButton.addActionListener(e -> syncComponentStatusWhenRadioButtonChanges());
+    }
+
+    @Override
+    protected @Nullable JComponent createCenterPanel() {
+        return rootPanel;
+    }
+
+    @Override
+    protected @NotNull List<ValidationInfo> doValidateAll() {
+        List<ValidationInfo> res = new ArrayList<>();
+        AuthConfiguration auth = getData();
+        if (StringUtils.isBlank(auth.getTenant())) {
+            res.add(new ValidationInfo("tenant is required.", tenantIdTextField));
+        }
+        if (!isGuid(auth.getTenant())) {
+            res.add(new ValidationInfo("tenant should be a guild.", tenantIdTextField));
```

Comment:
typo

Reply:
resolved in https://github.com/microsoft/azure-tools-for-java/pull/5265